File: frontend/src/types/analysis.ts
export type VitaminStatus = 'deficiency' | 'normal' | 'excess' | 'no_data'

export interface VitaminAnalysisItem {
  vitamin_id: number
  vitamin_name: string
  value: number | null
  unit: string
  norm_min: number
  norm_max: number
  status: VitaminStatus
  severity: number
}

export interface HistoryVitaminEntry {
  vitamin_id: number
  vitamin_name: string
  value: number
  unit: string
  status: VitaminStatus
}

export interface AnalysisSnapshot {
  date: string
  entries: HistoryVitaminEntry[]
}

export interface ComparisonItem {
  vitamin_name: string
  date1_value: number | null
  date2_value: number | null
  change_percent: number | null
  status1: string
  status2: string
}

File: backend/app/schemas/analysis.py
from typing import Optional, List
from pydantic import BaseModel


class VitaminAnalysisItem(BaseModel):
    vitamin_id: int
    vitamin_name: str
    value: Optional[float]
    unit: str
    norm_min: float
    norm_max: float
    status: str  # 'deficiency', 'normal', 'excess'
    severity: float  # percentage deviation from norm (0 = normal)


class HistoryVitaminEntry(BaseModel):
    vitamin_id: int
    vitamin_name: str
    value: float
    unit: str
    status: str

    class Config:
        from_attributes = True


class AnalysisSnapshot(BaseModel):
    date: str
    entries: List[HistoryVitaminEntry]


class ProductSearchResult(BaseModel):
    id: int
    name: str
    category: str
    vitamin_content: List["ProductVitaminContent"]

    class Config:
        from_attributes = True


class ProductVitaminContent(BaseModel):
    vitamin_id: int
    vitamin_name: str
    amount_per_100g: float
    unit: str


class ComparisonItem(BaseModel):
    vitamin_name: str
    date1_value: Optional[float] = None
    date2_value: Optional[float] = None
    change_percent: Optional[float] = None
    status1: str
    status2: str


ProductSearchResult.model_rebuild()
